atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -254,6 +254,20 @@ public int getDefaultLockTimeoutSeconds() {
         return AtlasDbConstants.DEFAULT_LOCK_TIMEOUT_SECONDS;
     }
 
+
+    /**
+     * The number of timestamps to cache that we have seen in previous reads.
+     * This will use somewhere around 90MB of heap memory per million timestamps because of various overheads
+     * from Java Objects and the cache's LRU tracking.
+     *
+     * Probably the only reason to configure away from the default would be a service that can afford the heap usage,
+     * and has read patterns that deal with a very large working set of existing transactions.
+     */
+    @Value.Default
+    public long getTimestampCacheSize() {
```

Comment:
Should this go in `AtlasDbRuntimeConfig` instead so you can resize it to larger/smaller live?

Reply:
It wouldn't be easy to make this live changeable; TimestampCache can't know what an AtlasRuntimeConfig is because of circular dependency problems, and you only initialize the parent SnapshotTransactionManager once.

Reply:
:/ I feel like we need to address the underlying issue. This makes it public api and yet another thing to break when we want the config / TMs stuff to not be such a mess.